presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -138,9 +161,11 @@ public ResourceGroupInfo getInfo()
                     new DataSize(softMemoryLimitBytes, BYTE),
                     maxRunningQueries,
                     maxQueuedQueries,
-                    runningQueries.size() + descendantRunningQueries,
-                    queuedQueries.size() + descendantQueuedQueries,
+                    canRunMore() ? CAN_RUN : (canQueueMore() ? CAN_QUEUE : BLOCKED),
+                    eligibleSubGroups.size(),
                     new DataSize(cachedMemoryUsageBytes, BYTE),
+                    infos.stream().flatMap(info -> info.getAggregatedRunningQueries().stream()).collect(Collectors.toList()),
```

Comment:
This looks like it creates a lot of redundant information (every node aggregates the queries below it). Is that necessary?